core/perf_test/PerfTestBlasKernels.hpp

```diff
@@ -165,12 +161,8 @@ struct DotSingle
   typedef typename Device::execution_space execution_space ;
 
   typedef typename
-    Impl::StaticAssertSame< Impl::unsigned_< 1 > ,
-                            Impl::unsigned_< Type::Rank > >::type ok_rank ;
-
-/*  typedef typename
-    Impl::StaticAssertSame< execution_space ,
-                            typename Type::execution_space >::type ok_device ;*/
+    std::is_same< Impl::unsigned_< 1 > ,
```

Comment:
Ditto.

Reply:
Fixed.

Reply:
Any reason `static_assert( Type::Rank == 1, "...");` would not work?

Reply:
`Rank` has type `int` except for Stokhos View specializations.  I usually just cast `Type::Rank` to `int` and everything works fine.